crates/primitives/src/bridge.rs

```diff
@@ -240,18 +251,39 @@ impl From<PubNonce> for Musig2PubNonce {
 
 impl BorshSerialize for Musig2PubNonce {
     fn serialize<W: Write>(&self, writer: &mut W) -> std::io::Result<()> {
-        writer.write_all(&self.0.serialize())?;
+        // Serialize self.0 (the PubNonce) into bytes
+        let nonce_bytes = self.0.serialize();
 
-        Ok(())
+        // Write the length prefix as a u32 (little-endian)
+        borsh::BorshSerialize::serialize(&(nonce_bytes.len() as u32), writer)?;
+
+        // Write the nonce bytes
+        writer.write_all(&nonce_bytes)
     }
 }
 
 impl BorshDeserialize for Musig2PubNonce {
     fn deserialize_reader<R: Read>(reader: &mut R) -> std::io::Result<Self> {
-        let mut nonce_bytes = [0u8; PUB_NONCE_SIZE];
+        // Read the length prefix as a u32 (little-endian)
+        let len = u32::deserialize_reader(reader)? as usize;
+
+        if len != PUB_NONCE_SIZE {
+            return Err(std::io::Error::new(
+                std::io::ErrorKind::InvalidData,
+                format!(
+                    "Invalid Musig2PubNonce size, expected: {}, got: {}",
+                    PUB_NONCE_SIZE, len
+                ),
+            ));
```

Comment:
Same here as before, the length tagging is completely redundant if we're confident the `PUB_NONCE_SIZE` is actually constant (which it is, right?).

Reply:
Done in https://github.com/alpenlabs/strata/commit/19c2bdb8c54554795da7a9404abd544bfe948ff2.